src/app/tests/TestReadInteraction.cpp

```diff
@@ -23,45 +23,148 @@
  */
 
 #include <app/InteractionModelEngine.h>
+#include <app/MessageDef/EventDataElement.h>
+#include <app/util/basic-types.h>
 #include <core/CHIPCore.h>
 #include <core/CHIPTLV.h>
 #include <core/CHIPTLVDebug.hpp>
 #include <core/CHIPTLVUtilities.hpp>
 #include <messaging/ExchangeContext.h>
 #include <messaging/ExchangeMgr.h>
 #include <messaging/Flags.h>
+#include <messaging/tests/MessagingContext.h>
+#include <nlunit-test.h>
 #include <platform/CHIPDeviceLayer.h>
 #include <protocols/secure_channel/MessageCounterManager.h>
 #include <protocols/secure_channel/PASESession.h>
 #include <support/ErrorStr.h>
 #include <support/UnitTestRegistration.h>
 #include <system/SystemPacketBuffer.h>
+#include <system/SystemTimer.h>
 #include <system/TLVPacketBufferBackingStore.h>
 #include <transport/SecureSessionMgr.h>
 #include <transport/raw/UDP.h>
+#include <transport/raw/tests/NetworkTestHelpers.h>
+#include <type_traits>
 
-#include <nlunit-test.h>
+namespace {
+chip::TransportMgrBase gTransportManager;
+chip::Test::LoopbackTransport gLoopback;
+chip::secure_channel::MessageCounterManager gMessageCounterManager;
+uint8_t gDebugEventBuffer[128];
+uint8_t gInfoEventBuffer[128];
+uint8_t gCritEventBuffer[128];
+chip::app::CircularEventBuffer gCircularEventBuffer[3];
+chip::NodeId kTestNodeId           = 1;
+chip::ClusterId kTestClusterId     = 6;
+chip::EndpointId kTestEndpointId   = 1;
+chip::EventId kTestEventIdDebug    = 1;
+chip::EventId kTestEventIdCritical = 1;
+using TestContext                  = chip::Test::MessagingContext;
+TestContext sContext;
+
+void InitializeEventLogging(chip::Messaging::ExchangeManager & aExchangeManager)
+{
+    chip::app::LogStorageResources logStorageResources[] = {
+        { &gDebugEventBuffer[0], sizeof(gDebugEventBuffer), nullptr, 0, nullptr, chip::app::PriorityLevel::Debug },
+        { &gInfoEventBuffer[0], sizeof(gInfoEventBuffer), nullptr, 0, nullptr, chip::app::PriorityLevel::Info },
+        { &gCritEventBuffer[0], sizeof(gCritEventBuffer), nullptr, 0, nullptr, chip::app::PriorityLevel::Critical },
+    };
 
-namespace chip {
-System::Layer gSystemLayer;
-SecureSessionMgr gSessionManager;
-Messaging::ExchangeManager gExchangeManager;
-TransportMgr<Transport::UDP> gTransportManager;
-const Transport::AdminId gAdminId = 0;
-secure_channel::MessageCounterManager gMessageCounterManager;
+    chip::app::EventManagement::CreateEventManagement(
+        &aExchangeManager, sizeof(logStorageResources) / sizeof(logStorageResources[0]), gCircularEventBuffer, logStorageResources);
```

Comment:
```suggestion
        &aExchangeManager, ArraySize(logStorageResources), gCircularEventBuffer, logStorageResources);
```
and fixing the other places that have that pattern (TestEventLogging.cpp, chip_im_responder.cpp), might be nice, as a followup.

Reply:
https://github.com/project-chip/connectedhomeip/issues/8372